mlir-tensorrt/executor/lib/Runtime/API/API.cpp

```diff
@@ -463,8 +466,6 @@ void AllocTracker::track(PointerInfo info) {
     map.insert(std::make_pair(info.ptr, std::move(value)));
     return;
   }
-  untrack(info.ptr);
-  map.insert(std::make_pair(info.ptr, std::move(value)));
```

Comment:
It's not clear to me why removing these lines is OK. The value struct that was previously in the map may be different than what is being passed into this function.

Reply:
Explained in the PR description.

Reply:
But there's still a problem by dropping these lines then, which is explained in the comment above starting on line 449. 

I think what we want here is something like `if( existingInfo.isExternallyManged && info.isExternallyManged ) return)`, otherwise retain the original behavior.